R/ospPlotConfiguration.R

```diff
@@ -0,0 +1,137 @@
+    # plotArea = NULL,
+    # panelArea = NULL,
+    # xGrid = NULL,
+    # yGrid = NULL,
+
+    #' @field watermark `tlf::Label` object defining watermark.
+    watermark = NULL,
+
+    #' @field lines `tlf::ThemeAestheticSelections` object or list defining how
+    #'   lines are plotted.
+    lines = NULL,
+
+    #' @field points `tlf::ThemeAestheticSelections` object or list defining how
+    #'   points are plotted.
+    points = NULL,
+
+    #' @field ribbons `tlf::ThemeAestheticSelections` object or list defining
+    #'   how ribbons are plotted.
+    ribbons = NULL,
+
+    #' @field errorbars `tlf::ThemeAestheticSelections` object or list defining
+    #'   how errorbars are plotted.
+    errorbars = NULL,
+
+    #' @field export R6 class `tlf::SaveConfiguration` defining saving properties.
+    export = NULL,
+
+    #' @description Create a new `ospPlotConfiguration` object
+    #'
+    #' @param title,subtitle,xlabel,ylabel A character string providing plot
+    #'   annotations for plot title, subtitle, x-axis label, y-axis label, plot
+    #'   legend, watermark, respectively.
+    #' @param legend `tlf::LegendConfiguration` object defining legend properties.
+    #' @param xAxis `tlf::XAxisConfiguration` object defining x-axis properties.
+    #' @param yAxis `tlf::YAxisConfiguration` object defining y-axis properties.
+    # @param background `tlf::BackgroundConfiguration` object defining background properties.
+    # @param plotArea `tlf::BackgroundElement` object defining properties of plot area.
+    # @param panelArea `tlf::BackgroundElement` object defining properties of panel area.
+    # @param xGrid `tlf::LineElement` object defining properties of x-grid background.
+    # @param yGrid `tlf::LineElement` object defining properties of y-grid background.
+    #' @param watermark `tlf::Label` object defining watermark.
+    #' @param lines `tlf::ThemeAestheticSelections` object or list defining how
+    #'   lines are plotted.
+    #' @param points `tlf::ThemeAestheticSelections` object or list defining how
+    #'   points are plotted.
+    #' @param ribbons `tlf::ThemeAestheticSelections` object or list defining
+    #'   how ribbons are plotted.
+    #' @param errorbars `tlf::ThemeAestheticSelections` object or list defining
+    #'   how errorbars are plotted.
+    #' @param export R6 class `tlf::SaveConfiguration` defining saving
+    #'   properties.
+    #'
+    #' @return A new `PlotConfiguration` object
+    initialize = function(title = NULL,
+                          subtitle = NULL,
+                          xlabel = NULL,
+                          ylabel = NULL,
+                          legend = NULL,
+                          xAxis = NULL,
+                          yAxis = NULL,
+                          # background = NULL,
+                          # plotArea = NULL,
+                          # panelArea = NULL,
+                          # xGrid = NULL,
+                          # yGrid = NULL,
```

Comment:
Not sure if we should be supporting these [parameters](https://www.open-systems-pharmacology.org/TLF-Library/reference/PlotConfiguration.html). 

![image](https://user-images.githubusercontent.com/11330453/161708027-4198d138-d49e-44c7-a757-09ddb5d8188f.png)

I feel these aspects of the plot are better specified through a _tlf_ theme, which will be another parameter for all plotting functions.

Reply:
I would say no.